crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -152,7 +152,12 @@ where
     }
 
     fn header_td_by_number(&self, number: BlockNumber) -> ProviderResult<Option<U256>> {
-        self.database.header_td_by_number(number)
+        if let Some(td) = self.database.header_td_by_number(number)? {
+            Ok(Some(td))
+        } else {
+            let last_persisted_block_number = self.database.last_block_number()?;
+            self.database.header_td_by_number(last_persisted_block_number)
```

Comment:
this still needs to check that we have the block in the canonical buffer otherwise this will blindly return the last written TD

so this should ensure that the canonical memory contains that block number, otherwise return None